aten/src/THCUNN/generic/RReLU.cu

```diff
@@ -16,7 +22,12 @@ void THNN_(RReLU_updateOutput)(
            void *generator)
 {
   THCUNN_assertSameGPU(state, 3, input, output, noise);
-  curandStateMtgp32* gen_states = THCRandom_generatorStates(state);
+  THCGenerator* gen = THCRandom_getGenerator(state);
+  // each thread will utilize one random, however, since we have to use
+  // curand_uniform4 (See Note [Register spilling in curand call for CUDA < 10]),
+  // offset is 4.
+  uint64_t offset = gen->state.philox_seed_offset.fetch_add(4);
```

Comment:
note that NUM_BLOCKS for most cases will be set to 64 (that's a poor choice, but for the next PR), so you'll have a grid-stride loop inside the kernel and generate multiple randoms, so adjust offset accordingly.

Reply:
updated offset calc with (numel / block_size * grid.x) * 4.